### PR TITLE
Initial tut docs.

### PR DESCRIPTION
**Pull request by sellout**
Still very spartan, but at least has actual Matryoshka content now.

Also adds some helpful new constructors for *ercusive instances.

### CODE REVIEW

File: docs/src/main/tut/docs/02-A-First-Example.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  case SubtractF(a, b) => "($a) - ($b)"
+  case MultiplyF(a, b) => "($a) * ($b)"
+  case DivideF(a, b)   => "($a) / ($b)"
+}
+```
+
+```tut
+val expr: Arithmetic =
+  Add(Multiply(Number(3), Divide(Number(4), Number(5))), Number(6))
+expr.cata(pprint)
```
Comment (sellout) [inline docs/src/main/tut/docs/02-A-First-Example.md]: When I `sbt tut`, it complains here that “value cata is not a member of Arithmetic”, but I _think_ I have all my imports right, and the implicits are at the top level now. Not sure what I’m missing.

Comment (sellout) [inline docs/src/main/tut/docs/02-A-First-Example.md]: There’s another file (03-Algebras.md) that also calls `cata`, and that works. The only difference I can think of is that that’s using a Recursive instance that’s not defined in the same file. Also very likely I just have a simple mistake elsewhere …

**Comment by rabbitonweb** [comment]
@sellout Beside tut docs I see some code changes as well. Just to be clear, those should also be part of this PR?

**Comment by sellout** [comment]
@rabbitonweb Yep – that’s why this is a “feature” bump rather than “revision” – they’re motivated by the docs, and mentioned in the PR description.

**Comment by sellout** [comment]
@rabbitonweb Another old PR for you to look at.

**Comment by sellout** [comment]
@rabbitonweb Ping.

**Comment by sellout** [comment]
@rabbitonweb Ping. This PR is going to be eligible for a drivers license soon.

**Comment by sellout** [comment]
I think this one will have to be updated after #79 is merged.

**Comment by alissapajer** [comment]
@rabbitonweb could you review this? If you don't want to review it, I'm happy to take it over. This has been pending since January.

**Comment by rabbitonweb** [comment]
I'm away from home till 3rd May but I will review it when my family is sleeping today. What about conflicting code? Ignore it for now?

**Comment by alissapajer** [comment]
Yeah, now that #79 and maybe some other PRs have landed, it looks like the docs need some updating. /cc @sellout regarding when to merge in master.

**Comment by sellout** [comment]
Yeah, I should probably merge first. Enough has changed (like being able to use `Birecursive` directly now), that it’ll affect the docs.

**Pull request merged**